tests/app/oidc/test_gcp_oidc.py

```diff
@@ -0,0 +1,65 @@
+import time
+
+from cachetools.func import ttl_cache
+from freezegun import freeze_time
+from mock import Mock, patch
+
+from app.oidc.gcp_oidc import TTL, OIDCCredentialsServiceGCP
+
+TEST_SDS_OAUTH2_CLIENT_ID = "TEST_SDS_OAUTH2_CLIENT_ID"
+
+
+@patch("app.oidc.gcp_oidc.Request")
+@patch("app.oidc.gcp_oidc.fetch_id_token_credentials")
+def test_oidc_credentials_service_gcp(mock_token_fetch, mock_request):
+    oidc_credentials_service = OIDCCredentialsServiceGCP()
+
+    # fetch credentials and check the gcp call is made
+    oidc_credentials_service.get_credentials(iap_client_id=TEST_SDS_OAUTH2_CLIENT_ID)
+    mock_token_fetch.assert_called_once_with(
+        audience=TEST_SDS_OAUTH2_CLIENT_ID, request=mock_request()
+    )
+    mock_token_fetch.return_value.refresh.assert_called_once_with(mock_request())
+
+
+@freeze_time("2023-01-01T12:00:00")
+@patch("app.oidc.gcp_oidc.Request", Mock)
+@patch("app.oidc.gcp_oidc.fetch_id_token_credentials")
+def test_oidc_credentials_service_gcp_ttl(mock_token_fetch):
```

Comment:
Minor, you don't need `gcp` in naming, the file is specifically for gcp. Can even drop `oidc_credentials_service` as this file is about that. My suggestion would be `test_get_credentials` and `test_get_credentials_ttl` or `test_ttl`.

Reply:
Sure thing, done